server/src/storage.rs

```diff
@@ -114,35 +111,30 @@ pub trait ObjectStorage: Sync + 'static {
             return Ok(());
         }
 
-        let entries = fs::read_dir(&CONFIG.parseable.local_disk_path)?
-            .map(|res| res.map(|e| e.path()))
-            .collect::<Result<Vec<_>, io::Error>>()?;
+        let streams = STREAM_INFO.list_streams();
 
-        for entry in entries {
-            let path = entry.into_os_string().into_string().unwrap();
-            let init_sync = StorageSync::new(path);
+        for stream in streams {
+            let sync = StorageSync::new(stream.clone());
 
-            let dir = init_sync.get_dir_name();
-
-            for file in WalkDir::new(&format!("{}/tmp", &dir.dir_name_local))
+            for file in WalkDir::new(sync.dir.temp_dir)
+                .min_depth(1)
+                .max_depth(1)
                 .into_iter()
                 .filter_map(|file| file.ok())
+                .map(|file| file.path().to_path_buf())
+                .filter(|file| file.is_file())
             {
-                if file.metadata().unwrap().is_file() {
-                    let file_local = format!("{}", file.path().display());
-                    let file_s3 = file_local.replace("/tmp", "");
-                    let final_s3_path =
-                        file_s3.replace(&format!("{}/", CONFIG.parseable.local_disk_path), "");
-                    let f_path = str::replace(&final_s3_path, ".", "/");
-                    let f_new_path = f_path.replace("/parquet", ".parquet");
-                    let _put_parquet_file = self.upload_file(&f_new_path, &file_local).await?;
-                    if let Err(e) = dir.delete_parquet_file(file_local.clone()) {
-                        log::error!(
-                            "Error deleting parquet file in path {} due to error [{}]",
-                            file_local,
-                            e
-                        );
-                    }
+                let filename = file.file_name().unwrap().to_str().unwrap();
+                let file_suffix = str::replacen(filename, ".", "/", 3);
```

Comment:
lets avoid adding new unwraps

Reply:
file is a valid utf-8 filename generated when local sync happens. Only way this could fail is if someone created an invalid file in the tmp directory.

Reply:
Someone could try this, we can't underestimate our users :)